src/proof/mod.rs

```diff
@@ -168,51 +166,172 @@ pub trait Provable<F: LurkField> {
     fn num_frames(&self) -> usize;
 }
 
```

Comment:
Nit : I'd love a one-liner comment here introducing folks to the 2 upcoming traits, telling them in effect that `RecursiveSNARKTrait` abstracts over 2 sorts of `Proof`, while `Prover` abstracts over 2 sorts of `Prover`s.